transport/tchannel/outbound.go

```diff
@@ -133,10 +133,14 @@ func callWithPeer(ctx context.Context, req *transport.Request, peer *tchannel.Pe
 		return nil, err
 	}
 
+	reqHeaders := req.Headers.Items()
+	if exactCaseHeader {
+		reqHeaders = req.Headers.ExactCaseItems()
+	}
 	// Inject tracing system baggage
-	reqHeaders := tchannel.InjectOutboundSpan(call.Response(), req.Headers.Items())
+	reqHeaders = tchannel.InjectOutboundSpan(call.Response(), reqHeaders)
```

Comment:
Now we've dropped canonicalization of the jaeger headers..

it seems like in the case where we are not using exact case headers, we should canonicalize the newly added jaeger headers like we had previously.